modules/mcc/include/opencv2/mcc/ccm.hpp

```diff
@@ -0,0 +1,537 @@
+Grayscale refers to the Y channel of the XYZ color space.
+The color space of the detected data is not determined and cannot be converted into the XYZ space.
+Therefore, the sRGB formula is used to approximate[5].
+\f[
+G_{s}=0.2126R_{s}+0.7152G_{s}+0.0722B_{s}
+\f]
+Then the polynomial parameters can be obtained by using the polyfit.
+\f[
+f=polyfit(G_{s}, G_{dl})
+\f]
+After \f$f\f$ is obtained, linearization can be performed.
+
+#### Logarithmic Polynomial Fitting
+
+For gamma correction formula, we take the logarithm:
+\f[
+ln(C_{sl})={\gamma}ln(C_s),\qquad C_s\ge0\
+\f]
+It can be seen that there is a linear relationship between \f$ln(C_s)\f$ and \f$ln(C_{sl})\f$. It can be considered that the formula is an approximation of a polynomial relationship, that is, there exists a polynomial $f$, which makes[2]:
+\f[
+ln(C_{sl})=f(ln(C_s)), \qquad C_s>0\\
+C_{sl}=0, \qquad C_s=0
+\f]
+
+Because \f$exp(ln(0))\to\infin\f$, the channel whose component is 0 is directly mapped to 0 in the formula above.
+
+For fitting channels respectively, we have:
+\f[
+r=polyfit(ln(R_s),ln(R_{dl}))\\
+g=polyfit(ln(G_s),ln(G_{dl}))\\
+b=polyfit(ln(B_s),ln(B_{dl}))\\
+\f]
+Note that the parameter of $ln$ cannot be 0.
+Therefore, we need to delete the channels whose values are 0 from $R_s$ and $R_{dl}$, $G_s$ and $G_{dl}$, $B_s$ and $B_{dl}$.
+
+Therefore:
+
+\f[
+ln(R_{sl})=r(ln(R_s)), \qquad R_s>0\\
+R_{sl}=0, \qquad R_s=0\\
+ln(G_{sl})=g(ln(G_s)),\qquad G_s>0\\
+G_{sl}=0, \qquad G_s=0\\
+ln(B_{sl})=b(ln(B_s)),\qquad B_s>0\\
+B_{sl}=0, \qquad B_s=0\\
+\f]
+
+For grayscale polynomials, there are also:
+\f[
+f=polyfit(ln(G_{sl}),ln(G_{dl}))
+\f]
+and:
+\f[
+ln(C_{sl})=f(ln(C_s)), \qquad C_s>0\\
+C_sl=0, \qquad C_s=0
+\f]
+*/
+enum LINEAR_TYPE
+{
+    IDENTITY_,
+    GAMMA,
+    COLORPOLYFIT,
+    COLORLOGPOLYFIT,
+    GRAYPOLYFIT,
+    GRAYLOGPOLYFIT
+};
+
+/** @brief Enum of possible functions to calculate the distance between colors.
+
+See https://en.wikipedia.org/wiki/Color_difference for details
+*/
+enum DISTANCE_TYPE
+{
+    CIE76,                      ///< The 1976 formula is the first formula that related a measured color difference to a known set of CIELAB coordinates.
+    CIE94_GRAPHIC_ARTS,         ///< The 1976 definition was extended to address perceptual non-uniformities.
+    CIE94_TEXTILES,
+    CIE2000,
+    CMC_1TO1,                   ///< In 1984, the Colour Measurement Committee of the Society of Dyers and Colourists defined a difference measure, also based on the L*C*h color model.
+    CMC_2TO1,
+    RGB,
+    RGBL
+};
+
+/** @brief Core class of ccm model
+
+Produce a ColorCorrectionModel instance for inference
+*/
+class CV_EXPORTS_W ColorCorrectionModel
+{
+public:
+    /** @brief Color Correction Model
+
+    Supported list of color cards:
+    - @ref Macbeth (Macbeth ColorChecker)
+    - @ref Vinyl (DKK ColorChecker)
+    - @ref DigitalSG (DigitalSG ColorChecker with 140 squares)
```

Comment:
Move "Supported list" out of params or just add reference to enum (use `@ref CONST_COLOR` for that).

Reply:
Thanks, I got it.